server/schedulers/balance_region.go

```diff
@@ -167,6 +171,13 @@ func (s *balanceRegionScheduler) Schedule(cluster opt.Cluster) []*operator.Opera
 			}
 			log.Debug("select region", zap.String("scheduler", s.GetName()), zap.Uint64("region-id", region.GetID()))
 
+			// Skip the empty region
+			if region.GetApproximateSize() <= core.EmptyRegionApproximateSize && regionCount > balanceEmptyRegionThreshold {
```

Comment:
it seems better to define a region filter and filter empty regions in `Rand{Pending, Follower, Leader, Learner}Region`.